api/include/opentelemetry/trace/trace_state.h

```diff
@@ -45,51 +45,149 @@ class TraceState
   class Entry
   {
   public:
-    Entry() noexcept = default;
-
-    // Copy constructor.
-    Entry(const Entry &copy);
-
-    // Creates an Entry for a given key-value pair.
-    Entry(nostd::string_view key, nostd::string_view value) noexcept;
-
-    nostd::string_view GetKey();
-    nostd::string_view GetValue();
+    Entry() : key_(nullptr), value_(nullptr){};
+
+    // Copy constructor
+    Entry(Entry &copy)
+    {
+      key_   = StringToPointer(nostd::string_view(copy.key_.get()));
+      value_ = StringToPointer(nostd::string_view(copy.value_.get()));
+    }
+
+    // Assignment operator
+    Entry &operator=(Entry &other)
+    {
+      key_   = StringToPointer(nostd::string_view(other.key_.get()));
+      value_ = StringToPointer(nostd::string_view(other.value_.get()));
+      return *this;
+    }
+
+    // Create an Entry for a given key-value pair.
+    Entry(nostd::string_view key, nostd::string_view value) noexcept
+    {
+      key_   = StringToPointer(key);
+      value_ = StringToPointer(value);
+    }
+
+    // Get the key associated with this entry.
+    nostd::string_view GetKey() { return nostd::string_view(key_.get()); }
+
+    // Get the value associated with this entry.
+    nostd::string_view GetValue() { return nostd::string_view(value_.get()); }
+
+    // Set the value for this entry. This overrides the previous value.
+    void SetValue(nostd::string_view value) { value_ = StringToPointer(value); }
 
   private:
     // Store key and value as raw char pointers to avoid using std::string.
     nostd::unique_ptr<const char[]> key_;
     nostd::unique_ptr<const char[]> value_;
+
+    // Copy string into a buffer and return a unique_ptr to the buffer.
+    // This is a workaround for the fact that strcpy doesn't accept a const char* destination.
+    nostd::unique_ptr<const char[]> StringToPointer(nostd::string_view str)
```

Comment:
Could we call this `CopyString`? So it's clear we're making a copy here, and not just converting from a string to a pointer.